openni2_launch/src/openni2_launch/wtf_plugin.py

```diff
@@ -0,0 +1,120 @@
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import print_function
+import logging
+import re
+import subprocess
+
+import rospy
+from roswtf.rules import warning_rule, error_rule
+
+
+def _device_notfound_subproc(id_manufacturer, id_product):
+    """
+    @rtype: [dict]
+    @return: Example:
+
+                        [{'device': '/dev/bus/usb/002/004', 'tag': 'Lenovo ', 'id': '17ef:305a'},
+                         {'device': '/dev/bus/usb/002/001', 'tag': 'Linux Foundation 3.0 root hub', 'id': '1d6b:0003'},
+                         {'device': '/dev/bus/usb/001/006', 'tag': 'Validity Sensors, Inc. ', 'id': '138a:0090'},,,]
+
+    @note: This method depends on Linux command (via subprocess), which makes
+                 this command platform-dependent. Ubuntu Xenial onward, a Python module
+                 that encapsulate platform operation becomes available so this method
+                 can be wiped out. See https://github.com/ros-drivers/openni2_camera/pull/80#discussion_r193295442
+    """
+    device_re = re.compile("Bus\s+(?P<bus>\d+)\s+Device\s+(?P<device>\d+).+ID\s(?P<id>\w+:\w+)\s(?P<tag>.+)$", re.I)
+    df = subprocess.check_output("lsusb")
+    devices = []
+    for i in df.split('\n'):
+        if i:
+            info = device_re.match(i)
+            if info:
+                dinfo = info.groupdict()
+                logging.debug("dinfo: {}, dinfo.id: {}".format(dinfo, dinfo["id"]))
+                if dinfo["id"] == "{}:{}".format(id_manufacturer, id_product):
+                    dinfo['device'] = "/dev/bus/usb/{}/{}".format(dinfo.pop('bus'), dinfo.pop('device'))
+                    devices.append(dinfo)
+    logging.info("#devices: {}\ndevices: {}".format(len(devices), devices))
+    return devices
+
+
+def sensor_notfound(ctx):
+    """
+    @summary: Check if expected number of sensors are found.
+              Expected number of sensors can be set by
+              ROS Parameter 'openni2_num_sensors_expected'.
+    @note: Technically this can be static check, but because of the
+           need for connecting to ROS Param server, this needs
+           to be online check.
+    """
+    errors = []
+    num_sensors_expected = rospy.get_param("openni2_num_sensors_expected", 1)
+    # TODO: The set of manufacture id and prod id is specific to Asus Xtion.
+    #       There may be other openni2-based devices.
+    devices = _device_notfound_subproc(
+        id_manufacturer="1d27", id_product="0601")
```

Comment:
I'm afraid hardcoding these values limits this feature to only Asus Xtion?

Reply:
Generalized in https://github.com/ros-drivers/openni2_camera/pull/80/commits/ff647b83d86cbcd57bba08fb6a5ac85a62386b98